prow/github/client.go

```diff
@@ -1347,6 +1351,87 @@ func (c *client) ListOrgInvitations(org string) ([]OrgInvitation, error) {
 	return ret, nil
 }
 
+// ListCurrentUserRepoInvitations lists pending invitations for the authenticated user.
+//
+// https://docs.github.com/en/rest/reference/repos#list-repository-invitations-for-the-authenticated-user
+func (c *client) ListCurrentUserRepoInvitations() ([]UserRepoInvitation, error) {
+	c.log("ListCurrentUserRepoInvitations")
+	if c.fake {
+		return nil, nil
+	}
+	path := "/user/repository_invitations"
+	var ret []UserRepoInvitation
+	err := c.readPaginatedResults(
+		path,
+		acceptNone,
+		"",
+		func() interface{} {
+			return &[]UserRepoInvitation{}
+		},
+		func(obj interface{}) {
+			ret = append(ret, *(obj.(*[]UserRepoInvitation))...)
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return ret, nil
+}
+
+// AcceptUserRepoInvitation accepts invitation for the authenticated user.
+//
+// https://docs.github.com/en/rest/reference/repos#accept-a-repository-invitation
+func (c *client) AcceptUserRepoInvitation(invitationID int) error {
+	c.log("AcceptUserRepoInvitation", invitationID)
+
+	_, err := c.request(&request{
+		method:    http.MethodPatch,
+		path:      fmt.Sprintf("/user/repository_invitations/%d", invitationID),
+		org:       "",
+		exitCodes: []int{204},
+	}, nil)
+
+	return err
+}
+
+func (c *client) ListCurrentUserOrgInvitations() ([]UserOrgInvitation, error) {
```

Comment:
We need to include the `state=pending` query parameter otherwise we will get both invitations and already active org memberships.

Reply:
See the docs here: https://docs.github.com/en/rest/reference/orgs#list-organization-memberships-for-the-authenticated-user
> Indicates the state of the memberships to return. Can be either active or pending. If not specified, the API returns both active and pending memberships.

We have a version of the pagination handler that handles query parameters. Please use that instead: https://github.com/kubernetes/test-infra/blob/ea8f981918d11fec80d037ad90768487ce18780a/prow/github/client.go#L1724-L1725

Reply:
cool, done.